src/test/java/com/puppycrawl/tools/checkstyle/JavaParserTest.java

```diff
@@ -205,8 +205,13 @@ public void testJava14TextBlocks() throws Exception {
     public void testNoStackOverflowOnDeepStringConcat() throws Exception {
         final File file =
                 new File(getPath("InputJavaParserNoStackOverflowOnDeepStringConcat.java"));
-        assertWithMessage("File parsing should not throw StackOverflowError.")
-                .that(JavaParser.parseFile(file, JavaParser.Options.WITH_COMMENTS))
+
+        final DetailAST ast = TestUtil.getResultWithLimitedResources(() -> {
+            return JavaParser.parseFile(file, JavaParser.Options.WITH_COMMENTS);
```

Comment:
Comment Only: If this is expanded, we will need to include more than just the `verify` methods.